test/fuzz/transformation_add_read_modify_write_atomic_instruction_test.cpp

```diff
@@ -0,0 +1,154 @@
+
+#include "gtest/gtest.h"
+#include "source/fuzz/fuzzer_util.h"
+#include "source/fuzz/instruction_descriptor.h"
+#include "test/fuzz/fuzz_test_util.h"
+
+namespace spvtools {
+namespace fuzz {
+namespace {
+
+// Draft WIP - NOT FINAL.
+TEST(TransformationAddReadModifyWriteAtomicInstructionTest, DISABLED_NotApplicable) {
+  const std::string shader = R"(
+               OpCapability Shader
+          %1 = OpExtInstImport "GLSL.std.450"
+               OpMemoryModel Logical GLSL450
+               OpEntryPoint Fragment %4 "main"
+               OpExecutionMode %4 OriginUpperLeft
+               OpSource ESSL 320
+          %2 = OpTypeVoid
+          %3 = OpTypeFunction %2
+          %6 = OpTypeInt 32 1g
+          %9 = OpTypeInt 32 0
+         %26 = OpTypeFloat 32 
+          %8 = OpTypeStruct %6
+         %10 = OpTypePointer StorageBuffer %8
+         %11 = OpVariable %10 StorageBuffer
+         %19 = OpConstant %26 0    
+         %18 = OpConstant %9 1
+         %12 = OpConstant %6 0
+         %13 = OpTypePointer StorageBuffer %6
+         %15 = OpConstant %6 4
+         %16 = OpConstant %6 7
+         %20 = OpConstant %9 80
+          %4 = OpFunction %2 None %3
+          %5 = OpLabel
+         %14 = OpAccessChain %13 %11 %12
+         %24 = OpAccessChain %13 %11 %12
+               OpReturn
+               OpFunctionEnd
+  )";
+
+  const auto env = SPV_ENV_UNIVERSAL_1_3;
+  const auto consumer = nullptr;
+  const auto context = BuildModule(env, consumer, shader, kFuzzAssembleOption);
+  spvtools::ValidatorOptions validator_options;
+  ASSERT_TRUE(fuzzerutil::IsValidAndWellFormed(context.get(), validator_options,
+                                               kConsoleMessageConsumer));
+  TransformationContext transformation_context(
+      MakeUnique<FactManager>(context.get()), validator_options);
+}
+
+// Draft WIP - NOT FINAL.
+TEST(TransformationAddReadModifyWriteAtomicInstructionTest, DISABLED_IsApplicable) {
+  const std::string shader = R"(
+               OpCapability Shader
+          %1 = OpExtInstImport "GLSL.std.450"
+               OpMemoryModel Logical GLSL450
+               OpEntryPoint Fragment %4 "main"
+               OpExecutionMode %4 OriginUpperLeft
+               OpSource ESSL 320
+          %2 = OpTypeVoid
+          %3 = OpTypeFunction %2
+          %6 = OpTypeInt 32 1g
+          %9 = OpTypeInt 32 0
+         %26 = OpTypeFloat 32 
+          %8 = OpTypeStruct %6
+         %10 = OpTypePointer StorageBuffer %8
+         %11 = OpVariable %10 StorageBuffer
+         %19 = OpConstant %26 0    
+         %18 = OpConstant %9 1
+         %12 = OpConstant %6 0
+         %13 = OpTypePointer StorageBuffer %6
+         %15 = OpConstant %6 4
+         %16 = OpConstant %6 7
+         %20 = OpConstant %9 80
+          %4 = OpFunction %2 None %3
+          %5 = OpLabel
+         %14 = OpAccessChain %13 %11 %12
+         %24 = OpAccessChain %13 %11 %12
```

Comment:
I suggest that you add some extra instructions in here, perhaps by writing a new shader in GLSL. It would be good to see some other instructions, that are not atomic instructions nor OpVariable nor OpAccessChain. Instructions that perform computation would be good. E.g. OpFAdd, OpIAdd, OpINotEqual, etc. 

Then you can apply transformations that add atomic RMW instructions above the computation instructions (OpFAdd, OpIAdd, OpINotEqual). I think this might make things a bit clearer for you when writing your transformation class, as you seem to be treating `instruction_to_insert_before` as some kind of atomic instruction, when it could in fact be almost any instruction.

Reply:
Done.